stdlib/src/utils/string_slice.mojo

```diff
@@ -26,6 +26,129 @@ alias StaticString = StringSlice[ImmutableStaticLifetime]
 """An immutable static string slice."""
 
 
+fn is_valid_utf8[
```

Comment:
Q: Where do you plan to use this? I'm asking because I want to be careful with adding public APIs, and if this is going to be mostly a string implementation detail, I'd rather call it `_is_valid_utf8` (especially since its signature is very unsafe).